Use node url from env variable for tests

vkgnosis opened the pull request:
We have a bunch of #ignore tests that are supposed to be run against a
real node. Instead of hardcoding the node url it is nicer to take it
from an env variable so that we don't have gnosis specific urls in the
code and so that it is easier to use a custom node.

### Test Plan
tested that env var works

vkgnosis committed:
Use node url from env variable for tests

We have a bunch of #ignore tests that are supposed to be run against a
real node. Instead of hardcoding the node url it is nicer to take it
from an env variable so that we don't have gnosis specific urls in the
code and so that it is easier to use a custom node.

nlordell commented on orderbook/src/account_balances.rs:
```diff
@@ -196,7 +196,7 @@ mod tests {
     #[ignore]
     async fn mainnet_can_transfer() {
-        let http = create_test_transport("http://127.0.0.1:8545");
+        let http = create_test_transport(&std::env::var("NODE_URL").unwrap());
```
nit: why not change the `create_test_transport` to just read from `env` directly instead of reading it at every call site?
```
fn create_test_transport() {
  let url = std::env::var("NODE_URL").unwrap()
  // ...
}
```

vkgnosis replied:
I felt that create_test_transport should just create the transport without deciding on the url. But I don't mind that way either.

nlordell replied:
Another suggestion would be to:
```
fn create_default_test_transport() {
  create_test_transport(std::env::var("NODE_URL").unwrap_or_else("http://127.0.0.1:8545"));
}
```

nlordell commented on solver/src/oneinch_solver.rs:
```diff
@@ -264,7 +265,7 @@ mod tests {
     #[tokio::test]
     #[ignore]
     async fn returns_error_on_non_mainnet() {
-        let web3 = testutil::infura("rinkeby");
+        let web3 = Web3::new(create_test_transport(&std::env::var("NODE_URL").unwrap()));
```
This test specifically needs to connect to a non-mainnet node in order to pass.

vkgnosis replied:
I understand that but imo you should set NODE_URL appropriately as you do for all the other tests.

nlordell replied:
But `testutil::infura()` already requires setting `INFURA_PROJECT_ID` implying that you are configuring the tests for use with Infura. My only concern with using `NODE_URL` here is that it makes it impossible to run both `solve_order_on_oneinch` and `returns_error_on_non_mainnet` in a single run.

vkgnosis replied:
On the other hand I can't run the tests if I'm not using infura.

nlordell replied:
Hmm, good point... What I don't like though is that `cargo test -p solver -- --ignored oneinch` to run all 1Inch solver tests (including the ignored ones) can never succeed. That being said, requiring Infura configuration for the test is also quite annoying.

Maybe `env::var::RINKEBY_NODE_URL`?

nlordell commented on solver/src/oneinch_solver.rs:
```diff
@@ -229,7 +230,7 @@ mod tests {
     #[tokio::test]
     #[ignore]
     async fn solve_order_on_oneinch() {
-        let web3 = testutil::infura("mainnet");
+        let web3 = Web3::new(create_test_transport(&std::env::var("NODE_URL").unwrap()));
```
This test specifically needs to connect to a mainnet node in order to pass.

bh2smith commented on shared/src/bad_token/trace_call.rs:
```diff
@@ -386,7 +386,7 @@ mod tests {
     #[ignore]
     async fn mainnet_tokens() {
         // shared::tracing::initialize("orderbook::bad_token=debug,shared::transport=debug");
-        let http = create_test_transport("https://dev-openethereum.mainnet.gnosisdev.com/");
```
I definitely think this is a good change - regardless of how it is implemented, because I was under the impression that this node url is not really meant to be made public.

nlordell reviewed (approved):
I **really** like this PR in general - I was just hoping to find a good solution to the particular of being able to run all 1Inch tests simultaneously.

vkgnosis commented:
I changed the oneinch test to use another env variable so it can be run in one command with the other.
Added another function that directly creates the transport from the env variable.

vkgnosis committed:
Merge branch 'main' into node_url
